### PR TITLE
Make sure we have a single code path on terminate.

### PR DESCRIPTION
**Pull request by obino**

### CODE REVIEW

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1088/

File: AppController/djinnServer.rb [hunk 1]
```diff
@@ -65,7 +65,7 @@ def on_init
     add_method(@djinn, "get_app_info_map", "secret")
     add_method(@djinn, "relocate_app", "appid", "http_port", "https_port",
       "secret")
-    add_method(@djinn, "kill", "secret")    
+    add_method(@djinn, "kill", "secret")
```
Comment (cdonati) [inline AppController/djinnServer.rb]: I think you need to add your new argument (stop_deployment) here.

File: AppController/terminate.rb [hunk 2]
```diff
@@ -18,10 +18,16 @@ def self.erase_appscale_state
     `service nginx reload`
     # Stop and then remove the service we configured with monit.
     `monit stop all`
+    while system("monit summary | grep Running > /dev/null") do
+      puts "Waiting for monit so stop services ..."
+      Kernel.sleep(2)
+    done
```
Comment (menivaitsi) [inline AppController/terminate.rb]: I believe this should be 'end' instead of 'done'.

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1090/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1092/

File: AppController/djinn.rb [hunk 3]
```diff
@@ -753,95 +753,43 @@ def relocate_app(appid, http_port, https_port, secret)
...
-              end
+    if my_node.is_login? and stop_deployment
+      Djinn.log_info("Stopping all other nodes.")
+      # Let's stop all other nodes.
+      threads << Thread.new {
+        @nodes.each { |node|
+          if node.private_ip != my_node.private_ip
+            acc = AppControllerClient.new(ip, @@secret)
+            begin
+              acc.kill()
```
Comment (cdonati) [inline AppController/djinn.rb]: I think this needs the stop_deployment argument.

**Comment by obino** [comment]
OCD https://ocd.appscale.com:8080/job/Daily%20Build/1095/
previous run got caught in between changes with the tools that created issues.

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1097/

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1099/

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1105/

Note: The AppController's kill path still isn't used since there are too many places where we rescue Exception, which catches a SIGTERM.

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1108/

**Pull request merged**